src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5429,9 +5438,28 @@ public override BoundNode VisitDiscardExpression(BoundDiscardExpression node)
 
         public override BoundNode VisitThrowExpression(BoundThrowExpression node)
         {
-            var result = base.VisitThrowExpression(node);
-            SetUnknownResultNullability();
-            return result;
+            VisitThrow(node.Expression);
```

Comment:
>VisitThrow [](start = 12, length = 10)

It looks like `_resultType` will be set to the `Exception` type which seems incorrect for a `throw` expression. The result type should probably be `node.Type` which is `null` (as you suggested using elsewhere).